c7n/resources/account.py

```diff
@@ -1386,3 +1386,54 @@ def process(self, resources):
             client.put_public_access_block(
                 AccountId=r['account_id'],
                 PublicAccessBlockConfiguration=config)
+
+
+@filters.register('glue-security-config')
+class GlueEncryptionEnabled(MultiAttrFilter):
+    """Check if security configuration is enabled on the aws account. """
+    retry = staticmethod(QueryResourceManager.retry)
+
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['glue-security-config']},
+            'CatalogEncryptionMode': {'type': 'string'},
+            'SseAwsKmsKeyId': {'type': 'string'},
+            'ReturnConnectionPasswordEncrypted': {'type': 'boolean'},
+            'AwsKmsKeyId': {'type': 'string'}}}
+
+    annotation = "c7n:glue-security-config"
+    permissions = ('glue:GetDataCatalogEncryptionSettings',)
+
+    def validate(self):
+        attrs = set()
+        for key in self.data:
+            if key in ['CatalogEncryptionMode',
+                       'ReturnConnectionPasswordEncrypted',
+                       'SseAwsKmsKeyId',
+                       'AwsKmsKeyId']:
+                attrs.add(key)
+        self.multi_attrs = attrs
+        return super(GlueEncryptionEnabled, self).validate()
+
+    def get_target(self, resource):
+        if self.annotation in resource:
+            return resource[self.annotation]
+        client = local_session(self.manager.session_factory).client('glue')
+        encryption_setting = client.get_data_catalog_encryption_settings().get(
+            'DataCatalogEncryptionSettings')
+        resource[self.annotation] = encryption_setting.get('EncryptionAtRest')
+        resource[self.annotation].update(encryption_setting.get('ConnectionPasswordEncryption'))
+
+        for x in self.data:
+            if 'alias' in self.data[x]:
```

Comment:
you can dedent this block with a not clause, but whats alias in this context.

Reply:
from what I was seeing in examples/tests it looked like alias key names were always prefixed with "alias" (ex: key: "alias/aws/ebs") so I was checking if alias was in the string to determine if an alias was being passed in or not.
Is there a better way to check if alias is being used as the key? I've seen c7n:AliasName being used to specify that but not all the time.

Reply:
Kapil, could you elaborate on the not clause. Not sure if I fully understand what you mean there

Reply:
this is pretty implicit code with not great variable naming, re for x in data.. this is evaluating 'type` key as well. given your only processing two

re not block you can dedent the entire thing which improves readability.. 

```
if not self.data[x].startswith('alias'):
   continue
```
at which point the rest of this code block dedents.